src/components/modal/examples/NestedDialog.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <HoppModal title="Dialog with Input field" @close="emit('modal-reject')">
+    <template #body>
+      <div>
+        <HoppButtonSecondary
+          label="Open Nested Dialog with Input"
+          @click="openNestedInputDialog"
+        />
+
+        <p>
+          <HoppSmartInput
+            type="text"
+            placeholder="Enter some text..."
+            v-model="text"
+            @submit="emit('modal-resolve', { text })"
+          />
+        </p>
+      </div>
+    </template>
+    <template #footer>
+      <HoppButtonPrimary
+        label="Submit from Parent"
+        @click="emit('modal-resolve', { text })"
+      />
+      <HoppButtonSecondary label="Cancel" @click="emit('modal-reject')" />
```

Comment:
I don't think we need a cancel button since we have the cancel icon at the top of the modal

Reply:
It's just for a demo. Moreover, cancel and close both have different meanings